doc/extensions/acrbuildext/azext_acrbuildext/build_task.py

```diff
@@ -0,0 +1,79 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import LongRunningOperation
+from ._utils import (
+    arm_deploy_template_build_task_create,
+    validate_managed_registry
+)
+from .build import _check_image_name
+
+BUILD_TASKS_NOT_SUPPORTED = 'Build Tasks are only supported for managed registries.'
+
+
+def acr_build_task_create(
+    cmd,
+    client,
+    build_task_name,
+    registry_name,
+    context,            
+    source_branch,  
```

Comment:
A few parameters could have default values such as `branch`. There are too many required parameters here. As a guideline 3-4 required parameters are at most (Powershell cmdlets enforce this). We had to make `location` optional in `acr create` for the same reason.